tests/openthread/main.c

```diff
@@ -0,0 +1,210 @@
+#include <stdio.h>
+
+#include "net/ipv6/addr.h"
+#include "openthread/ip6.h"
+#include "openthread/thread.h"
+#include "openthread/udp.h"
+#include "ot.h"
+#include "shell.h"
+#include "shell_commands.h"
+
+otUdpSocket mSocket;
+
+static OT_JOB _set_panid(otInstance *ot_instance, void *context)
+{
+    uint16_t panid = *((uint16_t *) context);
+
+    otLinkSetPanId(ot_instance, panid);
+}
+
+static OT_JOB _get_panid(otInstance *ot_instance, void *context)
+{
+    *((uint16_t *) context) = otLinkGetPanId(ot_instance);
+    printf("PanID: %04x\n", *((uint16_t *) context));
+}
+
+static OT_JOB _thread_start(otInstance *ot_instance, void *context)
+{
+    printf("Starting OpenThread\n");
+    otIp6SetEnabled(ot_instance, true);
+    otThreadSetEnabled(ot_instance, true);
+}
+
+static OT_JOB _read_state(otInstance *ot_instance, void *context)
+{
+    uint8_t state = otThreadGetDeviceRole(ot_instance);
+
+    switch (state) {
+        case kDeviceRoleOffline:
+            puts("offline");
+            break;
+        case kDeviceRoleDisabled:
+            puts("disabled");
+            break;
+        case kDeviceRoleDetached:
+            puts("detached");
+            break;
+        case kDeviceRoleChild:
+            puts("child");
+            break;
+        case kDeviceRoleRouter:
+            puts("router");
+            break;
+        case kDeviceRoleLeader:
+            puts("leader");
+            break;
+    }
+}
+
+static OT_JOB _get_ip_addresses(otInstance *ot_instance, void *context)
+{
+    for (const otNetifAddress *addr = otIp6GetUnicastAddresses(ot_instance); addr; addr = addr->mNext) {
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+        ipv6_addr_to_str(addrstr, (ipv6_addr_t *) &addr->mAddress.mFields, sizeof(addrstr));
+        printf("inet6 %s\n", addrstr);
```

Comment:
Is there a way to distinguish the different kind of addresses ? (link-local, RLOC, EID)